src/compass/s1_geocode_slc.py

```diff
@@ -167,68 +152,120 @@ def run(cfg: GeoRunConfig):
+            # outputs
+            phase_names = ['azimuth_carrier_phase', 'flattening_phase']
+            phase_descrs = [f'{pol} geocoded CSLC image {desc}'
+                            for desc in phase_names]
+
+            # Prepare arrays and datasets for carrier phase and flattening
+            # phase
+            ((carrier_phase_data_blk, carrier_phase_ds),
+             (flatten_phase_data_blk, flatten_phase_ds)) = \
+            [(np.full(out_shape, np.nan).astype(np.float64),
+                  init_geocoded_dataset(grid_group, ds_name, geo_grid,
+                                        np.float64, ds_desc,
+                                        output_cfg=cfg.output_params))
+                 for ds_name, ds_desc in zip(phase_names, phase_descrs)]
+
+            # Geocode
+            isce3.geocode.geocode_slc(geo_data_blocks=geo_data_blks,
+                                      rdr_data_blocks=rdr_data_blks,
+                                      dem_raster=dem_raster,
+                                      radargrid=radar_grid,
+                                      geogrid=geo_grid, orbit=orbit,
+                                      native_doppler=native_doppler,
+                                      image_grid_doppler=image_grid_doppler,
+                                      ellipsoid=ellipsoid,
+                                      threshold_geo2rdr=threshold,
+                                      num_iter_geo2rdr=iters,
+                                      sliced_radargrid=sliced_radar_grid,
+                                      first_azimuth_line=0,
+                                      first_range_sample=0,
+                                      flatten=flatten, reramp=True,
+                                      az_carrier=az_carrier_poly2d,
+                                      rg_carrier=isce3.core.Poly2d(np.array([0])),
+                                      az_time_correction=az_lut,
+                                      srange_correction=rg_lut,
+                                      carrier_phase_block=carrier_phase_data_blk,
+                                      flatten_phase_block=flatten_phase_data_blk)
+
+            # write geocoded data blocks to respective HDF5 datasets
+            geo_datasets.extend([carrier_phase_ds,
+                                               flatten_phase_ds])
+            geo_data_blks.extend([carrier_phase_data_blk,
+                                              flatten_phase_data_blk])
+            for cslc_dataset, cslc_data_blk in zip(geo_datasets,
+                                                   geo_data_blks):
+                # only convert/modify output if type not 'complex64'
+                # do nothing if type is 'complex64'
+                output_type = cfg.output_params.cslc_data_type
+                if output_type == 'complex32':
+                    cslc_data_blk = nisar.types.to_complex32(cslc_data_blk)
```

Comment:
```suggestion
                    cslc_data_blk = to_complex32(cslc_data_blk)
```
not defined here based on import